api/api.proto

```diff
@@ -3687,52 +3687,62 @@ message LocateResponse {
 }
 
 // VolumePlacementStrategy defines a strategy for placing volumes in the cluster which will be a series of rules
+// All the rules specified will be applied for volume placement.
+// Rules that have enforcement as "required" are strictly enforced while "preferred" are best effort.
+// In situations, where 2 or more rules conflict, the weight of the rules will dictate which wins.
 message VolumePlacementStrategy {
-  // Rules defines a list of rules as part of the placement spec. All the rules specified will
-  // be applied for volume placement.
-  // Rules that have enforcement as "required" are strictly enforced while "preferred" are best effort.
-  // In situations, where 2 or more rules conflict, the weight of the rules will dictate which wins.
-  repeated VolumePlacementRule rules = 1;
+  // ReplicaAffinity defines affinity rules between replicas within a volume
+  repeated ReplicaPlacementSpec replica_affinity = 1;
+  // ReplicaAntiAffinity defines anti-affinity rules between replicas within a volume
+  repeated ReplicaPlacementSpec replica_anti_affinity = 2;
```

Comment:
put the affinity type inside the ReplicaPlacementSpec and VolumePlacementSpec ?

Reply:
I kept it like this so that it's consistent with the k8s CRD. 

https://github.com/portworx/talisman/pull/88/files#diff-1f7c7c432e4b1db77dc8fa77eedcf8c2R40

`ReplicaPlacementSpec` here is a common type that can be used for affinity or anti-affinity.

When parsing, we will already know what it is using the field names replica_affinity and replica_anti_affinity.